### PR TITLE
Move examples, fix case normalization

### LINKED ISSUE #48: Extract out examples into a separate top-level "examples" project, and add more details
Currently thanks to @TobiasPfeifer and @ashprakasan, we have some examples of ZIO CLI in the `Example.scala`.

However, we should extract these examples into a separate top-level project, make sure each of them is in a separate file, and try to more fully flesh them out or add other examples that demonstrate more of ZIO CLI's features for options and arguments.

### PR DESCRIPTION
**Pull request by asakaev**
Closes #48

### CODE REVIEW

**Comment by asakaev** [comment]
@jdegoes

**Comment by asakaev** [comment]
@TobiasPfeifer

File: zio-cli/shared/src/main/scala/zio/cli/ParserOptions.scala [hunk 1]
```diff
@@ -23,5 +23,5 @@ object ParserOptions {
   /**
    * The default options are case sensitive parsing
    */
-  val default: ParserOptions = ParserOptions(true)
+  val default: ParserOptions = ParserOptions(false)
```
Comment (TobiasPfeifer) [inline zio-cli/shared/src/main/scala/zio/cli/ParserOptions.scala:26]: good spot 👍

**Review by TobiasPfeifer — APPROVED** [review]
👍  lgtm

**Comment by jdegoes** [comment]
@asakaev Thank you! Nice work.

It would be great if you could do a followup example that adds more details to the `git` example and makes it more realistic. Currently the examples are a bit toy and do not show users how to utilize the full range of ZIO CLI features.

**Pull request merged**